File: Python/503.py
class Solution(object):
    def nextGreaterElements(self, nums):
        lst = []
        for i in range(len(nums)):
            lst.append(loop(i,nums))
        return lst
def loop(i,nums):
    for j in range(i+1,len(nums)):
        if nums[j] > nums[i]:
            return nums[j]
    for j in range(0,i):
        if nums[j] > nums[i]:
            return nums[j]
    return -1

File: Java/503.java
class Solution {
    public int loop(int i,int[] nums){
        for(int j = i+1; j < nums.length; j++){
            if(nums[j] > nums[i]) return nums[j];
        }
        for(int j = 0; j < i; j++){
            if(nums[j] > nums[i]) return nums[j];
        }
        return -1;
    }
    public int[] nextGreaterElements(int[] nums) {
        int[] lst = new int[nums.length];
        for(int i = 0; i < nums.length; i++) lst[i] = loop(i,nums);
        return lst;
    }
}